enterprise/backend/src/metabase_enterprise/transforms/ordering.clj

```diff
@@ -56,24 +62,31 @@
 
   The result is a map of transform id -> #{transform ids the transform depends on}. Dependencies are limited to just
   the transforms in the original list -- if a transform depends on some transform not in the list, the 'extra'
-  dependency is ignored."
+  dependency is ignored. Both query and Python transforms can have dependencies on tables produced by other transforms."
   [transforms]
-  (let [transforms-by-db (->> transforms
+  (let [;; Group all transforms by their database
+        transforms-by-db (->> transforms
                               (map (fn [transform]
-                                     {(get-in transform [:source :query :database]) [transform]}))
+                                     (let [db-id (if (transforms.util/python-transform? transform)
+                                                   (get-in transform [:target :database])
```

Comment:
we should use `util/target-database-id` from this PR https://github.com/metabase/metabase/pull/63762/files#diff-7be56f5e5a3212803f9e96735b9893568c8d52504120046d83c5101dbc5e3e68R93